src/OrbitGl/TrackManager.cpp

```diff
@@ -170,15 +174,19 @@ void TrackManager::SetFilter(const std::string& filter) {
 void TrackManager::UpdateVisibleTrackList() {
   visible_tracks_.clear();
 
+  auto track_should_be_shown = [this](const Track* track) {
+    return track->GetVisible() && track_type_visibility_[track->GetType()];
```

Comment:
I like this, but could lead to some problems in the future. There could be a situation where we set ThreadTracks to visible and some thread track doesn't appear because their GetVisible returns false. I don't know what we should do here, all my ideas have pros and cons, but at least I want a bug to track this.

Reply:
True. I considered this, but in the end decided that this should be indicated by the UI if needed. My first approach was to simply change the `Visibility` of all tracks of a certain type, but I think this is too restrictive and I want to be able to control type visibility and track visibility individually, especially since I'm only planning on adding the type visibility UI in the first iteration. I am not 100% sure how I'd like both visibility settings to interact on UI side, but in the backend I think it's only positive to keep them separated.

I'd argue we should not create a bug for this because it's been a clear design decision and not something I was planning on changing later, wdyt?

Reply:
It's a future issue because currently we are not supporting hiding individual tracks. I still think that from the back-end, if we make an individual track visible, should be visible. But maybe you are right, and given that I don't have a really good proposal, we could leave this at it is.

Reply:
What do you think @beckerhe ?

Reply:
I tend to agree with Florian. It's good design to keep the visibility separate between Types and individual tracks, but that doesn't mean we need to expose that like that to the UI.

It's also not yet clear to me how hiding of individual tracks will be implemented. I'm not convinced that this will change the return value of `Track::GetVisible`. Maybe it's better to maintain an additional list of hidden tracks in `TrackManager` and `GetVisible` will only be used for logically hidden tracks.

Reply:
I was thinking of pulling `GetVisible()` into `CaptureViewElement` and then use it for explicitly hidden elements, I'd say that's the common pattern for UI elements. Right now, it has not been used at all, it wasn't even taken into account during rendering, so we can define it however we see fit :-)

But that's a change for another day anyways.